dotCMS/src/main/java/com/dotmarketing/common/reindex/ReIndexerStatus.java

```diff
@@ -0,0 +1,98 @@
+package com.dotmarketing.common.reindex;
+
+import com.dotcms.content.elasticsearch.business.ESIndexAPI;
+import java.io.Serializable;
+import java.util.Hashtable;
+import java.util.Map;
+
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotcms.content.elasticsearch.business.ContentletIndexAPIImpl;
+import com.dotcms.content.elasticsearch.business.IndiciesInfo;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+
+public class ReIndexerStatus implements Serializable {
+    private final ContentletIndexAPIImpl indexAPI;
+    private final ReindexQueueAPI queueAPI;
+    final ESIndexAPI esIndexAPI = APILocator.getESIndexAPI();
+
+    private ReIndexerStatus(final ContentletIndexAPIImpl indexAPI, final ReindexQueueAPI queueAPI) {
+        this.indexAPI = indexAPI;
+        this.queueAPI = queueAPI;
+    }
+
+
+    private static class ReIndexerStatusHolder {
+        static ReIndexerStatus statis = new ReIndexerStatus(new ContentletIndexAPIImpl(), APILocator.getReindexQueueAPI());
+
+    }
+
+    public ReIndexerStatus getInstance() {
+        return ReIndexerStatusHolder.statis;
+
+    }
+
+
+
+    public boolean inFullReindexation() throws DotDataException {
+        return indexAPI.isInFullReindex();
+    }
+
+    @CloseDBIfOpened
+    public static int getContentCountToIndex() throws DotDataException {
+
+        DotConnect dc = new DotConnect();
+        dc.setSQL("select count(*) as cc from contentlet_version_info");
+        return Integer.parseInt(dc.loadObjectResults().get(0).get("cc").toString());
+    }
+
+    @CloseDBIfOpened
+    public int getLastIndexationProgress() throws DotDataException {
+        return getLastIndexationProgress(getContentCountToIndex());
+    }
+
+    @CloseDBIfOpened
+    public int getLastIndexationProgress(int countToIndex) throws DotDataException {
+        long left = queueAPI.recordsInQueue();
+        int x = (int) (countToIndex - left);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'countToIndex' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)